docs/notebooks/documents/service_groups_manager.py

```diff
@@ -0,0 +1,62 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.16.1
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Service Groups Management for PDO Contracts
+#
+# This notebook helps to manage the services database.
+
+# %%
+import pdo.contracts.jupyter as pc_jupyter
+import IPython.display as ip_display
+import ipywidgets
+
+from pdo.contracts.services import service_labels
+from pdo.contracts.groups import ServiceGroupListWidget
+from pdo.contracts.groups import EnclaveServiceGroupCreateWidget
+from pdo.contracts.groups import ProvisioningServiceGroupCreateWidget
+from pdo.contracts.groups import StorageServiceGroupCreateWidget
+
+pc_jupyter.load_ipython_extension(get_ipython())
+
+try :
+    state
+except NameError:
+    (state, bindings) = pc_jupyter.initialize_environment('unknown')
+
+# %% [markdown]
+# ## List of Available Service Groups
+#
+# %%
+children = map(lambda stype : ServiceGroupListWidget(state, bindings, stype), service_labels.keys())
+service_list_widget = ipywidgets.Tab(children=list(children), titles=list(service_labels.values()))
+ip_display.display(service_list_widget)
+
+# %% [markdown]
+# ## Create a New Enclave Service Group
+#
+# %%
+ip_display.display(EnclaveServiceGroupCreateWidget(state, bindings))
+
+# %% [markdown]
+# ## Create a New Provisioning Service Group
+#
+# %%
+ip_display.display(ProvisioningServiceGroupCreateWidget(state, bindings))
+
+# %% [markdown]
+# ## Create a New Storage Service Group
+#
+# %%
+ip_display.display(StorageServiceGroupCreateWidget(state, bindings))
```

Comment:
(perhaps not a comment for this specific notebook:):

How do we ensure that the notebook user understands that the storage service group needs to be "compatible" with the eservice group when provisioning a contract? might be good to document this aspect somwhere.

Reply:
but it doesn't need to be compatible. the client will load the contract to the eservice. the storage services are exclusively for your replication protocol, NOT for use in eservices. that being said... a really smart/performant policy would connect the two. but there is no requirement that they must be the same.

Reply:
don't we need to assume that the storage group for sure includes sservices associated with the provisioned eservices? this would be the default assumption. We need to add more sservices to the storage group if the replication policy calls for additional redundancy.

Reply:
we do not.